lib/rules/no-positional-data-test-selectors.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const Rule = require('./base');
+const { builders: b } = require('ember-template-recast');
+
+module.exports = class NoPositionalDataTestSelectors extends Rule {
+  visitor() {
+    return {
+      MustacheStatement(node) {
+        this.process(node);
+      },
+      BlockStatement(node) {
+        this.process(node);
+      },
+    };
+  }
+
+  process(node) {
+    if (node.params && node.params.length) {
+      node.params.forEach((pair, index) => {
+        if (/data-+/.exec(pair.original)) {
+          if (this.mode === 'fix') {
+            node.params[index] = b.attr(pair.original, b.boolean('true'));
+          } else {
+            this.log({
+              message: `data-test selectors should have a default value. "${pair.original}" should be "${pair.original}=true"`,
+              line: node.loc && node.loc.start.line,
+              column: node.loc && node.loc.start.column,
+              source: this.sourceForNode(node),
+              isFixable: true,
+            });
+          }
+        }
+      });
+    }
```

Comment:
A few things;

* We shouldn't need to guard for the presence of `node.params`, every `MustacheStatement` and `BlockStatement` will have a params array
* We should be pushing a `HashPair` (not making the positional param value look like one)

```suggestion
    let testSelectorParamIndex = node.params.findIndex(
      (n) => n.type === "PathExpression" && n.original.startsWith("data-test-")
    );

    if (testSelectorParamIndex === -1) {
      return;
    }
    
    if (this.mode === 'fix') {
      let selectorName = node.params[testSelectorParamIndex].original;

      // remove the item from `params`
      node.params.splice(testSelectorParamIndex, 1);

      // add it as a HashPair
      node.hash.pairs.unshift(b.pair(selectorName, b.boolean(true)));
    } else {
      this.log({
        message: `Passing a `data-test-*` positional param to a curly invocation should be avoided.`,
        line: node.loc && node.loc.start.line,
        column: node.loc && node.loc.start.column,
        source: this.sourceForNode(node),
        isFixable: true,
      });
    }
```